crates/engine/tree/src/tree/root.rs

```diff
@@ -345,6 +343,7 @@ where
     let account_trie_nodes = proof_targets
         .into_par_iter()
         .map(|(hashed_address, hashed_slots)| {
+            let provider_ro = view.provider_ro().unwrap();
```

Comment:
I believe this can get expensive because we re-open the transaction for every account. Instead, we can use `map_init` that will delegate the on-demand creation of providers to Rayon.

Reply:
good call, done here https://github.com/paradigmxyz/reth/pull/12458/commits/38b2749131e46d61308ad57de279c1087ae8b624